lib/global-admin/addon/components/registry-catalog/component.js

```diff
@@ -0,0 +1,792 @@
+const REGISTRY_LIMIT_MEM =      'registry.registry.resources.limits.memory'
+
+const CLAIR_REQUEST_CPU =    'clair.resources.requests.cpu'
+const CLAIR_REQUEST_MEM =    'clair.resources.requests.memory'
+const CLAIR_LIMIT_CPU =      'clair.resources.limits.cpu'
+const CLAIR_LIMIT_MEM =      'clair.resources.limits.memory'
+
+const NOTARY_REQUEST_CPU =    'notary.resources.requests.cpu'
+const NOTARY_REQUEST_MEM =    'notary.resources.requests.memory'
+const NOTARY_LIMIT_CPU =      'notary.resources.limits.cpu'
+const NOTARY_LIMIT_MEM =      'notary.resources.limits.memory'
+
+const REGISTRY_NODE_SELECTOR_PREFIX  = 'registry.registry.nodeSelector."'
+const DATABASE_NODE_SELECTOR_PREFIX  = 'database.internal.nodeSelector."'
+const REDIS_NODE_SELECTOR_PREFIX     = 'redis.internal.nodeSelector."'
+const CLAIR_NODE_SELECTOR_PREFIX     = 'clair.nodeSelector."'
+const NOTARY_NODE_SELECTOR_PREFIX     = 'notary.nodeSelector."'
+
+const WORKLOADS = ['database', 'redis', 'registry', 'clair', 'notary']
+
+const HIDDEN_KEYS = {
+  'externalURL':         `https://${ window.location.host }`,
+  [INGRESS_HOSTS_CORE]:  window.location.host,
+  'persistence.type':    'storageClass',
+}
+
+const ANSWER_TO_CONFIG = {
+  secretKey:                'secretKey',
+  'imageChartStorage.type': 'storageType',
+  'database.type':          'databaseType',
+  'redis.type':             'redisType',
+  'clair.enabled':          'clairEnabled',
+  'notary.enabled':         'notaryEnabled',
+  'harborAdminPassword':    'password',
+  [DATABASE_REQUEST_CPU]:   'databaseRequestCpu',
+  [DATABASE_REQUEST_MEM]:   'databaseRequestMemory',
+  [DATABASE_LIMIT_CPU]:     'databaseLimitCpu',
+  [DATABASE_LIMIT_MEM]:     'databaseLimitMemory',
+  [REDIS_REQUEST_CPU]:      'redisRequestCpu',
+  [REDIS_REQUEST_MEM]:      'redisRequestMemory',
+  [REDIS_LIMIT_CPU]:        'redisLimitCpu',
+  [REDIS_LIMIT_MEM]:        'redisLimitMemory',
+  [REGISTRY_REQUEST_CPU]:   'registryRequestCpu',
+  [REGISTRY_REQUEST_MEM]:   'registryRequestMemory',
+  [REGISTRY_LIMIT_CPU]:     'registryLimitCpu',
+  [REGISTRY_LIMIT_MEM]:     'registryLimitMemory',
+  [CLAIR_REQUEST_CPU]:      'clairRequestCpu',
+  [CLAIR_REQUEST_MEM]:      'clairRequestMemory',
+  [CLAIR_LIMIT_CPU]:        'clairLimitCpu',
+  [CLAIR_LIMIT_MEM]:        'clairLimitMemory',
+  [NOTARY_REQUEST_CPU]:     'notaryRequestCpu',
+  [NOTARY_REQUEST_MEM]:     'notaryRequestMemory',
+  [NOTARY_LIMIT_CPU]:       'notaryLimitCpu',
+  [NOTARY_LIMIT_MEM]:       'notaryLimitMemory',
+}
+
+export default Component.extend(CrudCatalog, ReservationCheck, {
+  intl:        service(),
+  globalStore: service(),
+  growl:       service(),
+
+  layout,
+
+  answers:    null,
+  appName:    'global-registry',
+  nsName:     'cattle-system',
+  appVersion: APP_VERSION,
+  nsExists:   true,
+  config:     null,
+
+  level:      'cluster',
+
+  init() {
+    this._super(...arguments);
+    WORKLOADS.map((w) => {
```

Comment:
Since you've already defined a new function for the config can you move this into its own function as well?